flipperdevices/flipperzero-firmware
# [FL-2997] Improve file name filtering

# What's new

- Better definition of allowed characters in file names
- Helps to fix [this](https://github.com/flipperdevices/qFlipper/issues/133)

# Verification 

1. Compile and flash the firmware
2. Try writing a file containing one or more of the following characters in the name: `<>?*:|”\`. An error should be returned.
NOTE1: qFlipper ignores these errors, the file simply does not get written.
NOTE2: Might not be possible under Windows.
3. Try writing a file whose name does not contain the above characters (but contains other non-alphanumeric symbols). It should be written without problems both onto the internal and external storage.

# Checklist (For Reviewer)
